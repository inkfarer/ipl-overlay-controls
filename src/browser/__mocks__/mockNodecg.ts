import { Configschema } from '../../types/schemas';
import { NodeCGBrowser } from 'nodecg/browser';

const defaultBundleConfig = {
    lastfm: {
        apiKey: 'lastfmkey123',
        secret: 'lastfmsecret456'
    },
    smashgg: {
        apiKey: 'smashggkey789'
    },
    radia: {
        url: 'radia://url',
        socketUrl: 'ws://radia.url',
        authentication: 'radia-auth-12345'
    }
};

export const mockDialog = {
    open: jest.fn(),
    close: jest.fn()
};
export const mockGetDialog = jest.fn().mockReturnValue(mockDialog);
export const mockSendMessage = jest.fn();
export let replicants: {[key: string]: unknown} = {};
export let mockBundleConfig: Configschema = defaultBundleConfig;

beforeEach(() => {
    replicants = {};
    mockBundleConfig = defaultBundleConfig;
});

window.nodecg = {
    Replicant(name: string) {
        return {
            get value() {
                return replicants[name];
            },
            set value(newValue: unknown) {
                replicants[name] = newValue;
            },
            on: jest.fn()
        };
    },
    sendMessage: mockSendMessage,
    getDialog: mockGetDialog,
    bundleConfig: mockBundleConfig,
    bundleName: 'ipl-overlay-controls'
} as unknown as NodeCGBrowser;
