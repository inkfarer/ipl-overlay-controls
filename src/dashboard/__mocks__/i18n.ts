import i18next from 'i18next';
import { config } from '@vue/test-utils';

jest.mock('i18next-vue', () => ({
    __esModule: true,
    useTranslation: () => ({ t: i18next.t })
}));

beforeAll(async () => {
    await i18next.init({
        lng: 'cimode',
        appendNamespaceToCIMode: true
    });
    config.global.mocks = {
        $t: i18next.t
    };
});
