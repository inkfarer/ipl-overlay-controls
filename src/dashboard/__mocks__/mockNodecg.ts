import { Configschema } from 'schemas';
import NodeCG from '@nodecg/types';

const defaultBundleConfig = {
    lastfm: {
        apiKey: 'lastfmkey123',
        secret: 'lastfmsecret456'
    },
    smashgg: {
        apiKey: 'smashggkey789'
    },
    radia: {
        url: 'radia://url',
        socketUrl: 'ws://radia.url',
        authentication: 'radia-auth-12345'
    }
};

export const mockDialog = {
    open: jest.fn(),
    close: jest.fn()
};
export const mockGetDialog = jest.fn().mockReturnValue(mockDialog);
export const mockSendMessage = jest.fn();
export let replicants: {[key: string]: unknown} = {};
export let mockBundleConfig: Configschema = defaultBundleConfig;
export let messageListeners: {[key: string]: (message?: unknown, cb?: () => void) => void} = {};

beforeEach(() => {
    replicants = {};
    messageListeners = {};
    mockBundleConfig = defaultBundleConfig;
});

window.nodecg = {
    Replicant(name: string) {
        return {
            get value() {
                return replicants[name];
            },
            set value(newValue: unknown) {
                replicants[name] = newValue;
            },
            on: jest.fn()
        };
    },
    listenFor: (messageName: string, handler: () => void) => {
        messageListeners[messageName] = handler as (message: unknown, cb?: () => void) => void;
    },
    sendMessage: mockSendMessage,
    getDialog: mockGetDialog,
    bundleConfig: mockBundleConfig,
    bundleName: 'ipl-overlay-controls'
} as unknown as NodeCG.ClientAPI;
