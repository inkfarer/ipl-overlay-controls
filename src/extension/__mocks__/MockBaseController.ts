import { MessageInputMap, MessageResultMap } from '../../types/messages/MessageMaps';
import { BaseController } from '../controllers/BaseController';

export let controllerListeners:
    {[key in keyof MessageInputMap]?: (data?: unknown) => MessageResultMap[key] | Promise<MessageResultMap[key]>} = {};

beforeEach(() => {
    jest.spyOn(BaseController.prototype, 'listen').mockImplementation(
        <key extends keyof MessageInputMap>
        (name: key, callback: (data?: unknown) => any) => {
            controllerListeners[name] = callback;
        });

    controllerListeners = {};
});
