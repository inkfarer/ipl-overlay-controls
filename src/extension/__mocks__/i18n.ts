import i18next from 'i18next';

beforeAll(async () => {
    await i18next.init({
        lng: 'cimode',
        appendNamespaceToCIMode: true
    });
});
