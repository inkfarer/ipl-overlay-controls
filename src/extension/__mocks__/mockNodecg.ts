import express from 'express';
import last from 'lodash/last';
import { Configschema } from '../../types/schemas';
import cloneDeep from 'lodash/cloneDeep';

export type ReplicantChangeHandler = (newValue?: unknown, oldValue?: unknown) => void;
type OnFunction = (event: string, handler: ReplicantChangeHandler) => void;

const defaultBundleConfig = Object.freeze({
    lastfm: {
        apiKey: 'lastfmkey123',
        secret: 'lastfmsecret456'
    },
    smashgg: {
        apiKey: 'smashggkey789'
    },
    radia: {
        url: 'radia://url',
        socketUrl: 'ws://radia.url',
        authentication: 'radia-auth-12345'
    }
});

export let replicants: {[key: string]: unknown} = {};
export const replicantChangeListeners: {[key: string]: ReplicantChangeHandler} = {};
export const requestHandlers: {[type in 'POST' | 'GET']: {[path: string]: express.RequestHandler}}
    = { POST: {}, GET: {} };
export const mockBundleConfig: Configschema = cloneDeep(defaultBundleConfig);
export const messageListeners: {[key: string]: (message?: unknown, cb?: () => void) => void} = {};
export const mockMount = jest.fn();
export const mockNodecgLog = {
    warn: jest.fn(),
    info: jest.fn(),
    error: jest.fn()
};
export const mockSendMessage = jest.fn();

beforeEach(() => {
    replicants = {};
    Object.assign(mockBundleConfig, cloneDeep(defaultBundleConfig));
});

require('../helpers/nodecg').set({
    Replicant(name: string) {
        const replicantValue: { value: unknown, on: OnFunction } = {
            get value() {
                return replicants[name];
            },
            set value(newValue: unknown) {
                replicants[name] = newValue;
            },
            on: (event: string, handler: ReplicantChangeHandler) => {
                replicantChangeListeners[name] = handler;
            }
        };
        return replicantValue;
    },
    listenFor: (messageName: string, handler: () => void) => {
        messageListeners[messageName] = handler as (message: unknown, cb?: () => void) => void;
    },
    Router: () => ({
        post(path: string, ...handlers: express.RequestHandler[]) {
            requestHandlers['POST'][path] = last(handlers);
        },
        get(path: string, ...handlers: express.RequestHandler[]) {
            requestHandlers['GET'][path] = last(handlers);
        }
    }),
    mount: mockMount,
    log: mockNodecgLog,
    bundleConfig: mockBundleConfig,
    bundleName: 'ipl-overlay-controls',
    sendMessage: mockSendMessage
});
