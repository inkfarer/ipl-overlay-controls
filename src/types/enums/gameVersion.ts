export enum GameVersion {
    SPLATOON_2 = 'SPLATOON_2',
    SPLATOON_3 = 'SPLATOON_3'
}

export class GameVersionHelper {
    static toPrettyString(value: GameVersion): string {
        return {
            [GameVersion.SPLATOON_2]: 'Splatoon 2',
            [GameVersion.SPLATOON_3]: 'Splatoon 3'
        }[value];
    }
}
