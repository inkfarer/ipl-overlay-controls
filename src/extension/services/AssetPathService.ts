import type NodeCG from '@nodecg/types';
import { AssetPaths } from 'schemas';
import { GameVersion } from 'types/enums/gameVersion';
import { perGameData } from '../../helpers/gameData/gameData';

export class AssetPathService {
    private nodecg: NodeCG.ServerAPI;
    private assetPaths: NodeCG.ServerReplicant<AssetPaths>;

    constructor(nodecg: NodeCG.ServerAPI) {
        this.nodecg = nodecg;
        this.assetPaths = nodecg.Replicant<AssetPaths>('assetPaths', { persistent: false });
    }

    updateAssetPaths(gameVersion: GameVersion) {
        const gameData = perGameData[gameVersion];
        this.assetPaths.value.stageImages = Object.fromEntries(Object.entries(gameData.stageImagePaths)
            .map(([stage, path]) => ([stage, `/bundles/${this.nodecg.bundleName}/stage-images/${path}`])));
    }
}
