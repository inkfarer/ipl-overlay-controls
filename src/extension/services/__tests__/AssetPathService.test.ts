import { AssetPathService } from '../AssetPathService';
import { mockNodecg, replicants } from '../../__mocks__/mockNodecg';
import { GameVersion } from 'types/enums/gameVersion';
import { AssetPaths } from 'schemas';

describe('AssetPathService', () => {
    let assetPathService: AssetPathService;

    beforeEach(() => {
        assetPathService = new AssetPathService(mockNodecg);
        replicants.assetPaths = { stageImages: { } };
    });

    describe('updateAssetPaths', () => {
        it.each(Object.values(GameVersion))('updates asset paths as expected for %s', version => {
            assetPathService.updateAssetPaths(version);

            expect((replicants.assetPaths as AssetPaths).stageImages).toMatchSnapshot();
        });
    });
});