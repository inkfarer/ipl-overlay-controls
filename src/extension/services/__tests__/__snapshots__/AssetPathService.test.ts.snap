// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetPathService updateAssetPaths updates asset paths as expected for SPLATOON_2 1`] = `
{
  "Ancho-V Games": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Ancho-V_Games.webp",
  "Arowana Mall": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Arowana_Mall.webp",
  "Blackbelly Skatepark": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Blackbelly_Skatepark.webp",
  "Camp Triggerfish": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Camp_Triggerfish.webp",
  "Goby Arena": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Goby_Arena.webp",
  "Humpback Pump Track": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Humpback_Pump_Track.webp",
  "Inkblot Art Academy": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Inkblot_Art_Academy.webp",
  "Kelp Dome": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Kelp_Dome.webp",
  "MakoMart": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_MakoMart.webp",
  "Manta Maria": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Manta_Maria.webp",
  "Moray Towers": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Moray_Towers.webp",
  "Musselforge Fitness": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Musselforge_Fitness.webp",
  "New Albacore Hotel": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_New_Albacore_Hotel.webp",
  "Piranha Pit": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Piranha_Pit.webp",
  "Port Mackerel": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Port_Mackerel.webp",
  "Shellendorf Institute": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Shellendorf_Institute.webp",
  "Shifty Station": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Shifty_Station.webp",
  "Skipper Pavilion": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Skipper_Pavilion.webp",
  "Snapper Canal": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Snapper_Canal.webp",
  "Starfish Mainstage": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Starfish_Mainstage.webp",
  "Sturgeon Shipyard": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Sturgeon_Shipyard.webp",
  "The Reef": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_The_Reef.webp",
  "Wahoo World": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Wahoo_World.webp",
  "Walleye Warehouse": "/bundles/ipl-overlay-controls/stage-images/splatoon2/S2_Stage_Walleye_Warehouse.webp",
}
`;

exports[`AssetPathService updateAssetPaths updates asset paths as expected for SPLATOON_3 1`] = `
{
  "Barnacle & Dime": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Barnacle_&_Dime.webp",
  "Brinewater Springs": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Brinewater_Springs.webp",
  "Eeltail Alley": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Eeltail_Alley.webp",
  "Flounder Heights": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Flounder_Heights.webp",
  "Hagglefish Market": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Hagglefish_Market.webp",
  "Hammerhead Bridge": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Hammerhead_Bridge.webp",
  "Humpback Pump Track": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Humpback_Pump_Track.webp",
  "Inkblot Art Academy": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Inkblot_Art_Academy.webp",
  "Mahi-Mahi Resort": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Mahi_Mahi_Resort.webp",
  "MakoMart": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_MakoMart.webp",
  "Manta Maria": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Manta_Maria.webp",
  "Mincemeat Metalworks": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Mincemeat_Metalworks.webp",
  "Museum d'Alfonsino": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Museum_dAlfonsino.webp",
  "Scorch Gorge": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Scorch_Gorge.webp",
  "Sturgeon Shipyard": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Sturgeon_Shipyard.webp",
  "Um'ami Ruins": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Umami_Ruins.webp",
  "Undertow Spillway": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Undertow_Spillway.webp",
  "Wahoo World": "/bundles/ipl-overlay-controls/stage-images/splatoon3/S3_Stage_Wahoo_World.webp",
}
`;
